source/extensions/access_loggers/BUILD

```diff
@@ -1,16 +1,18 @@
 load(
     "//bazel:envoy_build_system.bzl",
-    "envoy_cc_library",
+    "envoy_cc_extension_library",
     "envoy_package",
 )
 
 licenses(["notice"])  # Apache 2
 
 envoy_package()
```

Comment:
We can define a `envoy_extension_package` with stricter `default_visibility`, use that instead of `envoy_package` in extensions. That limit visibility beyond cc_library.